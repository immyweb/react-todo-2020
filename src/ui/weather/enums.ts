export enum Locations {
  WICK = 'WICK',
  ABERDEEN = 'ABERDEEN',
  EDINBURGH = 'EDINBURGH',
  BELFAST = 'BELFAST',
  MANCHESTER = 'MANCHESTER',
  CARDIFF = 'CARDIFF',
  PLYMOUTH = 'PLYMOUTH',
  LONDON = 'LONDON',
}

export enum Days {
  MON = 'MON',
  TUE = 'TUE',
  WED = 'WED',
  THU = 'THU',
  FRI = 'FRI',
  SAT = 'SAT',
  SUN = 'SUN',
}

export enum Direction {
  LEFT = 'LEFT',
  RIGHT = 'RIGHT',
  TOP = 'TOP',
  BOTTOM = 'BOTTOM',
}

export enum WeatherIconUrls {
  CLEAR_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/clear-night@2x1.png',
  SUNNY_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png',
  PARTLY_CLOUDY_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Party-Cloudy.png',
  PARTLY_CLOUDY_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Party-Cloudy.png',
  MIST = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/foggy@2x1.png',
  FOG = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/foggy@2x1.png',
  CLOUDY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Cloudy.png',
  OVERCAST = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Cloudy.png',
  LIGHT_RAIN_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Rain2.png',
  LIGHT_RAIN_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Rain2.png',
  DRIZZLE = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Rain2.png',
  LIGHT_RAIN = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Rain2.png',
  HEAVY_RAIN_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Rain2.png',
  HEAVY_RAIN_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Rain2.png',
  HEAVY_RAIN = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Rain2.png',
  SLEET_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  SLEET_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  SLEET = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  HAIL_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Hail.png',
  HAIL_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Hail.png',
  HAIL = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Hail.png',
  LIGHT_SNOW_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  LIGHT_SNOW_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  LIGHT_SNOW = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  HEAVY_SNOW_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  HEAVY_SNOW_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  HEAVY_SNOW = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  THUNDER_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Storm.png',
  THUNDER_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Storm.png',
  THUNDER = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Storm.png',
}

export enum WeatherAtmosphereUrls {
  CLEAR_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_clearnight@2x.jpg',
  SUNNY_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_sunny@2x.jpg',
  PARTLY_CLOUDY_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_partlycloudynight@2x.jpg',
  PARTLY_CLOUDY_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_partlycloudynight@2x.jpg',
  MIST = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_foggy@2x.jpg',
  FOG = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_foggy@2x.jpg',
  CLOUDY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_cloudy@2x.jpg',
  OVERCAST = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_cloudy@2x.jpg',
  LIGHT_RAIN_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  LIGHT_RAIN_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  DRIZZLE = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  LIGHT_RAIN = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  HEAVY_RAIN_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  HEAVY_RAIN_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  HEAVY_RAIN = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  SLEET_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  SLEET_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  SLEET = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  HAIL_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  HAIL_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  HAIL = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  LIGHT_SNOW_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  LIGHT_SNOW_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  LIGHT_SNOW = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  HEAVY_SNOW_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  HEAVY_SNOW_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  HEAVY_SNOW = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  THUNDER_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_stormy@2x.jpg',
  THUNDER_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_stormy@2x.jpg',
  THUNDER = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_stormy@2x.jpg',
}
