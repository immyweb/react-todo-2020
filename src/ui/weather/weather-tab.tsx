import * as React from 'react';

import { colours } from '../../styles/colours';

import {
  WeatherTabContainer,
  WeatherTabBody,
  WeatherTabDay,
  WeatherTabDate,
  WeatherTabArrow,
  WeatherTabTemperature,
  WeatherIconTab,
} from './styles';

export interface IProps {
  day: string;
  date: string;
  icon: string;
  dayMax: string;
  nightMin: string;
  selected?: boolean;
  onClick?: () => void;
}

export const WeatherTab: React.SFC<IProps> = ({
  day,
  date,
  icon,
  dayMax,
  nightMin,
  selected = false,
  onClick,
}) => {
  const [r, g, b] = colours.$offBlackRGB;
  const opacity = selected ? '1' : '0.4';
  const backgroundColour = `rgba(${r}, ${g}, ${b}, ${opacity})`;

  return (
    <WeatherTabContainer onClick={onClick}>
      <WeatherTabArrow
        className="WeatherTabArrow"
        visible={!selected}
        backgroundColour={backgroundColour}
      />
      <WeatherTabBody
        className="WeatherTabBody"
        selected={selected}
        backgroundColour={backgroundColour}
      >
        <WeatherTabDay className="WeatherTabDay" colour={colours.$white}>
          {day}
          <WeatherTabDate>{`, ${date}`}</WeatherTabDate>
        </WeatherTabDay>
        <WeatherIconTab className="WeatherTabIcon" iconUrl={icon} />
        <WeatherTabTemperature className="WeatherTabTemperature" colour={colours.$white}>
          {`${dayMax}° | `}
          <span style={{ color: colours.$news }}>{`${nightMin}°`}</span>
        </WeatherTabTemperature>
      </WeatherTabBody>
    </WeatherTabContainer>
  );
};
