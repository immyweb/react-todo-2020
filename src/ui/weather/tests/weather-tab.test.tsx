import * as React from 'react';
import { shallow } from 'enzyme';
import { WeatherTab } from '../weather-tab';
import { WeatherIconUrls } from '../enums';

const defaultWeatherTabProps = {
  day: 'SAT',
  date: 'AUG 10',
  icon: WeatherIconUrls.SUNNY_DAY,
  dayMax: '22',
  nightMin: '5',
};

describe('<WeatherTab />', () => {
  it('renders correctly with default values', () => {
    const wrapper = shallow(<WeatherTab {...defaultWeatherTabProps} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders correctly with selected false', () => {
    const wrapper = shallow(<WeatherTab {...defaultWeatherTabProps} selected={true} />);
    expect(wrapper).toMatchSnapshot();
  });
});
