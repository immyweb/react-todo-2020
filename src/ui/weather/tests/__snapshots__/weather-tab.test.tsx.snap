// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherTab /> renders correctly with default values 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    date="AUG 10"
    day="SAT"
    dayMax="22"
    icon="https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png"
    nightMin="5"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <glamorous(div)
          backgroundColour="rgba(34, 37, 38, 0.4)"
          className="WeatherTabArrow"
          visible={true}
        />,
        <glamorous(div)
          backgroundColour="rgba(34, 37, 38, 0.4)"
          className="WeatherTabBody"
          selected={false}
        >
          <glamorous(div)
            className="WeatherTabDay"
            colour="#ffffff"
          >
            SAT
            <glamorous(div)>
              , AUG 10
            </glamorous(div)>
          </glamorous(div)>
          <glamorous(glamorous(div))
            className="WeatherTabIcon"
            iconUrl="https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png"
          />
          <glamorous(div)
            className="WeatherTabTemperature"
            colour="#ffffff"
          >
            22° | 
            <span
              style={
                Object {
                  "color": "#4085cb",
                }
              }
            >
              5°
            </span>
          </glamorous(div)>
        </glamorous(div)>,
      ],
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "backgroundColour": "rgba(34, 37, 38, 0.4)",
          "className": "WeatherTabArrow",
          "visible": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "backgroundColour": "rgba(34, 37, 38, 0.4)",
          "children": Array [
            <glamorous(div)
              className="WeatherTabDay"
              colour="#ffffff"
            >
              SAT
              <glamorous(div)>
                , AUG 10
              </glamorous(div)>
            </glamorous(div)>,
            <glamorous(glamorous(div))
              className="WeatherTabIcon"
              iconUrl="https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png"
            />,
            <glamorous(div)
              className="WeatherTabTemperature"
              colour="#ffffff"
            >
              22° | 
              <span
                style={
                  Object {
                    "color": "#4085cb",
                  }
                }
              >
                5°
              </span>
            </glamorous(div)>,
          ],
          "className": "WeatherTabBody",
          "selected": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "SAT",
                <glamorous(div)>
                  , AUG 10
                </glamorous(div)>,
              ],
              "className": "WeatherTabDay",
              "colour": "#ffffff",
            },
            "ref": null,
            "rendered": Array [
              "SAT",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": ", AUG 10",
                },
                "ref": null,
                "rendered": ", AUG 10",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "WeatherTabIcon",
              "iconUrl": "https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "22° | ",
                <span
                  style={
                    Object {
                      "color": "#4085cb",
                    }
                  }
                >
                  5°
                </span>,
              ],
              "className": "WeatherTabTemperature",
              "colour": "#ffffff",
            },
            "ref": null,
            "rendered": Array [
              "22° | ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "5°",
                  "style": Object {
                    "color": "#4085cb",
                  },
                },
                "ref": null,
                "rendered": "5°",
                "type": "span",
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <glamorous(div)
            backgroundColour="rgba(34, 37, 38, 0.4)"
            className="WeatherTabArrow"
            visible={true}
          />,
          <glamorous(div)
            backgroundColour="rgba(34, 37, 38, 0.4)"
            className="WeatherTabBody"
            selected={false}
          >
            <glamorous(div)
              className="WeatherTabDay"
              colour="#ffffff"
            >
              SAT
              <glamorous(div)>
                , AUG 10
              </glamorous(div)>
            </glamorous(div)>
            <glamorous(glamorous(div))
              className="WeatherTabIcon"
              iconUrl="https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png"
            />
            <glamorous(div)
              className="WeatherTabTemperature"
              colour="#ffffff"
            >
              22° | 
              <span
                style={
                  Object {
                    "color": "#4085cb",
                  }
                }
              >
                5°
              </span>
            </glamorous(div)>
          </glamorous(div)>,
        ],
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "backgroundColour": "rgba(34, 37, 38, 0.4)",
            "className": "WeatherTabArrow",
            "visible": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "backgroundColour": "rgba(34, 37, 38, 0.4)",
            "children": Array [
              <glamorous(div)
                className="WeatherTabDay"
                colour="#ffffff"
              >
                SAT
                <glamorous(div)>
                  , AUG 10
                </glamorous(div)>
              </glamorous(div)>,
              <glamorous(glamorous(div))
                className="WeatherTabIcon"
                iconUrl="https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png"
              />,
              <glamorous(div)
                className="WeatherTabTemperature"
                colour="#ffffff"
              >
                22° | 
                <span
                  style={
                    Object {
                      "color": "#4085cb",
                    }
                  }
                >
                  5°
                </span>
              </glamorous(div)>,
            ],
            "className": "WeatherTabBody",
            "selected": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "SAT",
                  <glamorous(div)>
                    , AUG 10
                  </glamorous(div)>,
                ],
                "className": "WeatherTabDay",
                "colour": "#ffffff",
              },
              "ref": null,
              "rendered": Array [
                "SAT",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": ", AUG 10",
                  },
                  "ref": null,
                  "rendered": ", AUG 10",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "WeatherTabIcon",
                "iconUrl": "https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "22° | ",
                  <span
                    style={
                      Object {
                        "color": "#4085cb",
                      }
                    }
                  >
                    5°
                  </span>,
                ],
                "className": "WeatherTabTemperature",
                "colour": "#ffffff",
              },
              "ref": null,
              "rendered": Array [
                "22° | ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "5°",
                    "style": Object {
                      "color": "#4085cb",
                    },
                  },
                  "ref": null,
                  "rendered": "5°",
                  "type": "span",
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<WeatherTab /> renders correctly with selected false 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    date="AUG 10"
    day="SAT"
    dayMax="22"
    icon="https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png"
    nightMin="5"
    selected={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <glamorous(div)
          backgroundColour="rgba(34, 37, 38, 1)"
          className="WeatherTabArrow"
          visible={false}
        />,
        <glamorous(div)
          backgroundColour="rgba(34, 37, 38, 1)"
          className="WeatherTabBody"
          selected={true}
        >
          <glamorous(div)
            className="WeatherTabDay"
            colour="#ffffff"
          >
            SAT
            <glamorous(div)>
              , AUG 10
            </glamorous(div)>
          </glamorous(div)>
          <glamorous(glamorous(div))
            className="WeatherTabIcon"
            iconUrl="https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png"
          />
          <glamorous(div)
            className="WeatherTabTemperature"
            colour="#ffffff"
          >
            22° | 
            <span
              style={
                Object {
                  "color": "#4085cb",
                }
              }
            >
              5°
            </span>
          </glamorous(div)>
        </glamorous(div)>,
      ],
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "backgroundColour": "rgba(34, 37, 38, 1)",
          "className": "WeatherTabArrow",
          "visible": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "backgroundColour": "rgba(34, 37, 38, 1)",
          "children": Array [
            <glamorous(div)
              className="WeatherTabDay"
              colour="#ffffff"
            >
              SAT
              <glamorous(div)>
                , AUG 10
              </glamorous(div)>
            </glamorous(div)>,
            <glamorous(glamorous(div))
              className="WeatherTabIcon"
              iconUrl="https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png"
            />,
            <glamorous(div)
              className="WeatherTabTemperature"
              colour="#ffffff"
            >
              22° | 
              <span
                style={
                  Object {
                    "color": "#4085cb",
                  }
                }
              >
                5°
              </span>
            </glamorous(div)>,
          ],
          "className": "WeatherTabBody",
          "selected": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "SAT",
                <glamorous(div)>
                  , AUG 10
                </glamorous(div)>,
              ],
              "className": "WeatherTabDay",
              "colour": "#ffffff",
            },
            "ref": null,
            "rendered": Array [
              "SAT",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": ", AUG 10",
                },
                "ref": null,
                "rendered": ", AUG 10",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "WeatherTabIcon",
              "iconUrl": "https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "22° | ",
                <span
                  style={
                    Object {
                      "color": "#4085cb",
                    }
                  }
                >
                  5°
                </span>,
              ],
              "className": "WeatherTabTemperature",
              "colour": "#ffffff",
            },
            "ref": null,
            "rendered": Array [
              "22° | ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "5°",
                  "style": Object {
                    "color": "#4085cb",
                  },
                },
                "ref": null,
                "rendered": "5°",
                "type": "span",
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <glamorous(div)
            backgroundColour="rgba(34, 37, 38, 1)"
            className="WeatherTabArrow"
            visible={false}
          />,
          <glamorous(div)
            backgroundColour="rgba(34, 37, 38, 1)"
            className="WeatherTabBody"
            selected={true}
          >
            <glamorous(div)
              className="WeatherTabDay"
              colour="#ffffff"
            >
              SAT
              <glamorous(div)>
                , AUG 10
              </glamorous(div)>
            </glamorous(div)>
            <glamorous(glamorous(div))
              className="WeatherTabIcon"
              iconUrl="https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png"
            />
            <glamorous(div)
              className="WeatherTabTemperature"
              colour="#ffffff"
            >
              22° | 
              <span
                style={
                  Object {
                    "color": "#4085cb",
                  }
                }
              >
                5°
              </span>
            </glamorous(div)>
          </glamorous(div)>,
        ],
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "backgroundColour": "rgba(34, 37, 38, 1)",
            "className": "WeatherTabArrow",
            "visible": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "backgroundColour": "rgba(34, 37, 38, 1)",
            "children": Array [
              <glamorous(div)
                className="WeatherTabDay"
                colour="#ffffff"
              >
                SAT
                <glamorous(div)>
                  , AUG 10
                </glamorous(div)>
              </glamorous(div)>,
              <glamorous(glamorous(div))
                className="WeatherTabIcon"
                iconUrl="https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png"
              />,
              <glamorous(div)
                className="WeatherTabTemperature"
                colour="#ffffff"
              >
                22° | 
                <span
                  style={
                    Object {
                      "color": "#4085cb",
                    }
                  }
                >
                  5°
                </span>
              </glamorous(div)>,
            ],
            "className": "WeatherTabBody",
            "selected": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "SAT",
                  <glamorous(div)>
                    , AUG 10
                  </glamorous(div)>,
                ],
                "className": "WeatherTabDay",
                "colour": "#ffffff",
              },
              "ref": null,
              "rendered": Array [
                "SAT",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": ", AUG 10",
                  },
                  "ref": null,
                  "rendered": ", AUG 10",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "WeatherTabIcon",
                "iconUrl": "https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "22° | ",
                  <span
                    style={
                      Object {
                        "color": "#4085cb",
                      }
                    }
                  >
                    5°
                  </span>,
                ],
                "className": "WeatherTabTemperature",
                "colour": "#ffffff",
              },
              "ref": null,
              "rendered": Array [
                "22° | ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "5°",
                    "style": Object {
                      "color": "#4085cb",
                    },
                  },
                  "ref": null,
                  "rendered": "5°",
                  "type": "span",
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
